### ISSUE TITLE
[BUG] AirGapped Harvester UI Extension Keeps Mentioning Repository is missing but it's not

### ISSUE THREAD

**Issue by irishgordo** [harvester/harvester#7215]
### Describe The Bug:
When dealing with an airgapped registry serving `ui-plugin-catalog` & a repository that was an `airgapped` mirror - once Harvester UI extension is installed from AirGap, it seems to keep mentioning repository is missing.

### To Reproduce:

**Prerequisite [Hauler](https://docs.hauler.dev/docs/intro) on a Separate VM (Ubuntu Cloud Img) (all commands run from /home/ubuntu directory w/ ubuntu user):**
1. `curl -sfL https://get.hauler.dev | bash`
2. `hauler store add image docker.io/rancher/ui-plugin-catalog:3.1.0`
3. `wget https://github.com/minio/certgen/releases/download/v1.3.0/certgen-linux-amd64`
4. `mv certgen-linux-amd64 certgen`
5. `chmod +x certgen`
6. `sudo mv -v certgen /usr/local/bin/`
7. build a file in `/etc/systemd/system/hauler.service` that has the following content:
```bash
[Unit]
Description=Hauler Serve Registry Service

[Service]
Environment="HOME=/home/ubuntu/"
ExecStart=/usr/local/bin/hauler store serve registry --config /home/ubuntu/hauler_config.yaml
WorkingDirectory=/home/ubuntu

[Install]
WantedBy=multi-user.target
```
8. `sudo systemctl enable rc-local.service`
9. build a file in `/etc/rc.local` with the following content:
```
#!/bin/sh
# add your commands
# call your scripts here

/usr/sbin/ip link set enp1s0 up
/usr/sbin/dhclient -v enp1s0
# last line must be exit 0
exit 
```
10. `sudo chmod a+x /etc/rc.local && hauler store add image docker.io/rancher/ui-plugin-catalog:3.1.0`
11. reboot the vm / editing it's config / airgapping it's network
12. build this file in `/home/ubuntu/hauler_config.yaml` with the following content be sure to replace your ipv4 with the chunk ($REPLACE-WITH-YOUR-AIRGAP-IPV4):
```
version: 0.1
http:
  addr: $REPLACE-WITH-YOUR-AIRGAP-IPV4:5000
  tls:
    certificate: /home/ubuntu/public.crt
    key: /home/ubuntu/private.key
storage:
  filesystem:
    rootdirectory: /home/ubuntu/registry
    maxthreads: 100
```
13. replace $IPV4 with your airgapped IPV4, when you run this: `certgen -host "127.0.0.1,localhost,$IPV4,harvester-ui-airgap-registry.$IPV4.sslip.io"`
14. `sudo systemctl enable hauler.service`
15. `sudo systemctl start hauler.service`
16. validate it's working, replace $IPV4 with yours: `curl -k https://harvester-ui-airgap-registry.$IPV4.sslip.io:5000/v2/rancher/ui-plugin-catalog/tags/list | jq`

**Prerequisite GitLab Community Edition Mirror Of harvester-ui-extension (Ubuntu Cloud Img VM):**
1. `git clone --recursive https://github.com/harvester/harvester-ui-extension.git`
2. build a file in `/etc/rc.local` with the following content:
```
#!/bin/sh
# add your commands
# call your scripts here

/usr/sbin/ip link set enp1s0 up
/usr/sbin/dhclient -v enp1s0
# last line must be exit 0
exit 0
```
3. `sudo chmod a+x /etc/rc.local`
4. `sudo systemctl enable rc-local.service`
5. install docker: https://docs.docker.com/engine/install/ubuntu/
6. do docker post install: https://docs.docker.com/engine/install/linux-postinstall/
7. log out if needed ^ post install usually can be happier that way, then log back in
8. `docker pull gitlab/gitlab-ce:17.2.0-ce.0`
9. `sudo mkdir -p /srv/gitlab/config /srv/gitlab/logs /srv/gitlab/data`
10. restart & airgap your vm network
11. build a docker-compose file `/home/ubuntu/docker-compose.yaml` that holds (replace $IPV4 with yours):
```yaml
version: '3.6'
services:
  gitlab:
    image: gitlab/gitlab-ce:17.2.0-ce.0
    container_name: gitlab
    restart: always
    hostname: "$IPV4"
    environment:
      GITLAB_HOME: "/srv/gitlab"
      GITLAB_ROOT_PASSWORD: "p@ssword1234"
      GITLAB_ROOT_EMAIL: "root@local"
      EXTERNAL_URL: http://$IPV4
      GITLAB_OMNIBUS_CONFIG: |
        # Email setup
        gitlab_rails['gitlab_email_enabled'] = false
        gitlab_rails['smtp_enable'] = false
        gitlab_rails['smtp_enable_starttls'] = false
        # HTTPS Setup
        letsencrypt['enable'] = false
        gitlab_rails['gitlab_https'] = false
        gitlab_rails['gitlab_port'] = 80
    ports:
      - '80:80'
      - '443:443'
      - '2222:22'
    volumes:
      - '/srv/gitlab/config:/etc/gitlab'
      - '/srv/gitlab/logs:/var/log/gitlab'
      - '/srv/gitlab/data:/var/opt/gitlab'
    networks:
      - gitlab-network
    shm_size: '256m'

networks:
  gitlab-network:
    name: gitlab-network
```
12. `docker compose up -d`
13. log in with `root@local` & `p@ssword1234` to the GitLab instance running on the airgapped VM
14. build a new repository in gitlab, make it "public", could be `/root/`  then for slug `harvester-ui-extension`
15. copy the repo info, add as a remote ex: `cd /home/ubuntu/harvester-ui-extension && git remote add local http://$IPV4/root/harvester-ui-extension`
16. make git happier, `git config --global user.name "Administrator" && git config --global user.emal "root@local"`
17. push tags, `git push --set-upstream local --tags` , if prompted for email/pw fill out 
18. push gh-pages, `git checkout gh-pages && git push --set-upstream local`
19. airgapped gitlab commmunity edition with harvester-ui-extension repo should be live and working

**Prerequisite RKE2 or K3s:**
1. be sure to add your airgapped registry, like with k3s as a `registries.yaml` in `/etc/rancher/k3s/registries.yaml` with content like, replace $IPV4 with your airgap registries:
```yaml
mirrors:
  harvester-ui-airgap-registry.$IPV4.sslip.io:
    endpoint:
      - "https://harvester-ui-airgap-registry.$IPV4.sslip.io:5000"
configs:
  "harvester-ui-airgap-registry.$IPV4.sslip.io:5000":
    tls:
      insecure_skip_verify: true
```
2. `systemctl restart k3s.service` or rke2 
3. build a manifest on your host machine `manifest-for-ui-catalog-thing.yaml` replace $IPV4 with your airgap registries:
```yaml
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: ui-plugin-catalog
  namespace: cattle-ui-plugin-system
  labels:
    catalog.cattle.io/ui-extensions-catalog-image: ui-plugin-catalog
spec:
  replicas: 1
  selector:
    matchLabels:
      catalog.cattle.io/ui-extensions-catalog-image: ui-plugin-catalog
  template:
    metadata:
      namespace: cattle-ui-plugin-system
      labels:
        catalog.cattle.io/ui-extensions-catalog-image: ui-plugin-catalog
    spec:
      containers:
      - name: server
        image: harvester-ui-airgap-registry.$IPV4.sslip.io/rancher/ui-plugin-catalog:3.1.0
        imagePullPolicy: IfNotPresent
---
apiVersion: v1
kind: Service
metadata:
  name: ui-plugin-catalog-svc
  namespace: cattle-ui-plugin-system
spec:
  ports:
    - name: catalog-svc-port
      port: 8080
      protocol: TCP
      targetPort: 8080
  selector:
    catalog.cattle.io/ui-extensions-catalog-image: ui-plugin-catalog
  type: ClusterIP
---
apiVersion: catalog.cattle.io/v1
kind: ClusterRepo
metadata:
  name: ui-plugin-catalog-repo
spec:
  url: http://ui-plugin-catalog-svc.cattle-ui-plugin-system:8080
---
```
4. `kubectl create -f manifest-for-ui-catalog-thing.yaml` 

**Have Harvester v1.4.0 Cluster & Rancher v2.10.1:**
1. log in to Rancher v2.10.1
2. manage repositories from UI extensions, add your airgapped repo mirror ex: `A.B.C.D/root/harvester-ui-extension` and branch `gh-pages` 
3. add `v1.0.2` Harvester UI Extension
4. reload
5. import v1.4.0 cluster
6. observe that v1.4.0 cluster while it imports just fine, there's an issue with Rancher still thinking that there's no `harvester-ui-extensions` 

### Expected Behavior
The message not be shown, as we are currently & successfully using a `harvester-ui-extension` that did load all of it's .tgz assets for v1.0.2 -> it shouldn't be still "searching for a Harvester UI Extension to exist"

### Environment
 - Harvester ISO version: v1.4.0
 - Rancher Version: v2.10.1
 - Underlying Infrastructure: bare-metal hp dl360 gen 9

### Additional Context
![Screenshot from 2024-12-19 19-24-41](https://github.com/user-attachments/assets/7112f85d-5eaf-4e51-9cbb-3654d45e9e5a)
![Screenshot from 2024-12-19 18-43-52](https://github.com/user-attachments/assets/7dfa643f-315a-4b23-9e8b-34dbb18a3a13)
![Screenshot from 2024-12-19 18-40-57](https://github.com/user-attachments/assets/a109564f-5c93-4656-955e-5740ac58ecfa)

**Comment by torchiaf** [comment]
@irishgordo previous steps are all good.
The UI shows missing repo error because we are checking if the [harvester](https://github.com/harvester/harvester-ui-extension) catalog is imported, but for air-gapped environments it's clearly not the default one, so it doesn't recognize the air-gapped env registry as the correct one.
However, It should be always possible to install the extension manually.

Thanks for reporting this issue, we will fix it ASAP.

**Comment by momesgin** [comment]
@torchiaf A Jira ticket is also created, since I couldn't edit the description of this issue, I add the Jira reference here: SURE-9766

**Comment by torchiaf** [comment]
@irishgordo Here is the Rancher GH issues:
- 2.11 - https://github.com/rancher/dashboard/issues/13199
- 2.10.2 - https://github.com/rancher/dashboard/issues/13203

**Comment by a110605** [comment]
The fixed PRs have merged to Rancher 2.11.0 and 2.10.2. 
 
Hi @irishgordo or @TachunLin , could you use Rancher 2.10.2 to verify this issue in air-gapped env ?

**Comment by irishgordo** [comment]
@a110605 @FrankYang0529 
This looks good 😄 👍 

With Harvester v1.4.2 & v2.10.2 Rancher w/ Rancher UI Plugin Catalog: 
- https://hub.docker.com/layers/rancher/ui-plugin-catalog/3.4.3/images/sha256-310940f56846755baae13ea99332e52d713b9a1f1abcfbe60b4f3587f8cadd62

Utilizing [Hauler Version: v1.2.1](https://docs.hauler.dev/docs/guides-references/hauler-manifests) as underlying Registry/Store/Syncing.

With Manifest containing elements like:
```yaml
apiVersion: content.hauler.cattle.io/v1
kind: Images
metadata:
  annotations:
    hauler.dev/platform: linux/amd64
  name: important-image-stuff
spec:
  images:
# A LOT OF OTHER IMAGES
  - name: rancher/ui-plugin-catalog:3.3.0
```

![Image](https://github.com/user-attachments/assets/66e6c900-cf36-44c5-adba-67925db372db)
![Image](https://github.com/user-attachments/assets/17c27290-4b99-4081-8ba2-0716a65dafc7)
![Image](https://github.com/user-attachments/assets/d8686553-9803-4548-8b9c-9c92802de355)

**Comment by irishgordo** [comment]
@a110605 @FrankYang0529 
I'm re-opening this.

This is present on:
- Harvester v1.4.2
- Rancher v2.10.4 (that was previously upgraded from Rancher v2.9.6)
- Harvester UI Extension: v1.0.5 or v1.0.4
- Rancher UI Plugin Catalog: https://hub.docker.com/r/rancher/ui-plugin-catalog/tags , `4.0.0`

![Image](https://github.com/user-attachments/assets/af9fa0fe-da69-464a-8891-a195c38ccba4)

**Comment by a110605** [comment]
> [@a110605](https://github.com/a110605) [@FrankYang0529](https://github.com/FrankYang0529) I'm re-opening this.
> 
> This is present on:
> 


We didn't check the installation from image catalog that the repo is missing messages are pop up.
I open https://github.com/rancher/dashboard/issues/14141 to implement in Rancher v2.12.0.

Not sure we can close this ticket for v1.5.0 ? 

cc @innobead , @irishgordo

**Comment by innobead** [comment]
Added `require/external-rancher-update` to know this is an issue needed to be fixed at Rancher, upstream side. 

cc @harvester/tpm

**Comment by irishgordo** [comment]
To note, this is also present in:
- Harvester: v1.5.0-rc4
- Rancher: v2.11.0
- Harvester UI Extension: v1.5.0-rc4
- Harvester's Forked UI Plugin Catalog: rancher/ui-extension-harvester-ui-extension:1.5.0-rc4

![Image](https://github.com/user-attachments/assets/aec8a240-5551-40fa-9249-fd565475273b)

**Comment by a110605** [comment]
@innobead , @irishgordo , @houhoucoop , as we needs https://github.com/rancher/dashboard/issues/14141 to fix in Rancher 2.12.

Can we close this ticket and use https://github.com/rancher/dashboard/issues/14141 to follow up?

**Comment by starbops** [comment]
We usually keep the original ticket so we don't lose ends from our side. Since the actual fix is schedule in Rancher v2.12, I'll move this one to the v1.6.0 milestone first.

**Comment by innobead** [comment]
JFYI. We also have https://github.com/orgs/rancher/projects/103 to track Rancher upstream issues including UI.

cc @harvester/tpm @harvester/release

**Comment by irishgordo** [comment]
@a110605 This is not dependent on AirGap either.

Here's a loadout:
- Rancher v2.11.0
- Harvester UI Extension v1.5.0
- Harvester v1.5.0

Installed via Helm
```
helm install rancher rancher-latest/rancher  \
  --version 2.11.0 \
  --namespace cattle-system \
  --set hostname=rancher.10.115.3.10.sslip.io \
  --set global.cattle.psp.enabled=false \
  --set bootstrapPassword=testtesttest \
  --set replicas=1 \
  --debug
```


But, then update that Rancher on the K3s via Helm to Rancher v2.11.1 w/ HotFix tag:

```
helm upgrade rancher rancher-prime/rancher --set rancherImageTag=v2.11.1-hotfix-edbc.1  \
  --version 2.11.1 \
  --namespace cattle-system \
  --set hostname=rancher.10.115.3.10.sslip.io \
  --set global.cattle.psp.enabled=false \
  --set bootstrapPassword=testtesttest \
  --set replicas=1 \
  --debug
```

The same issue is presented.
So AirGap doesn't matter as much I don't think.
cc: @starbops 

![Image](https://github.com/user-attachments/assets/11974088-5c61-42ee-aa2a-c7e0600b7483)

**Comment by a110605** [comment]
We need to https://github.com/rancher/dashboard/issues/14141 to fix in Rancher 2.12 or 2.13.

**Comment by irishgordo** [comment]
As another mention, this is still present in:
- Rancher 2.10.4 w/ Harvester v1.4.2
- Rancher 2.11.2 w/ Harvester v1.4.2

Utilizing:
- Harvester UI Extension 1.0.5
- W/ Deployment Image being (Import Extension Catalog):
    - rancher/ui-plugin-catalog:4.0.1 , image

**Comment by Vicente-Cheng** [comment]
I followed the https://github.com/rancher/dashboard/issues/14141 and it seems we need to wait for rancher v2.13.
So, moved it to v1.7.0

**Comment by a110605** [comment]
https://github.com/rancher/dashboard/issues/14141 is already fixed via https://github.com/rancher/dashboard/pull/15044 and backport to Rancher 2.12.1.

**Comment by irishgordo** [comment]
Blocked by: https://github.com/harvester/harvester/issues/9209

**Comment by irishgordo** [comment]
Blocked by: https://github.com/rancher/rancher-prime/pull/335
cc: @Vicente-Cheng @khushboo-rancher 
Will re-evaluate tomorrow

**Comment by irishgordo** [comment]
Added dependency-blocked label, we need to wait for Rancher v2.13.0-alpha6, to be published and rancher-prime PR 335 to be merged in -> both are needed for the pre-release index so we can get the list to airgap all the images with Harvester.
This is also blocking any additional airgap validation prep currently.

**Comment by irishgordo** [comment]
I'm thinking things look good with alpha6, now:
```
╭─mike at suse-workstation-team-harvester in ~/Projects/harvester-cloud on doc/sample-doc-on-rke2-manifests-with-harvester-configuration-customization✘✘✘
╰─± skopeo list-tags docker://stgregistry.suse.com/rancher/turtles | grep -ie "v0.25.0-rc.4"
        "v0.25.0-rc.4",
        "v0.25.0-rc.4-linux-amd64",
        "v0.25.0-rc.4-linux-arm64"
╭─mike at suse-workstation-team-harvester in ~/Projects/harvester-cloud on doc/sample-doc-on-rke2-manifests-with-harvester-configuration-customization✘✘✘
╰─± skopeo list-tags docker://stgregistry.suse.com/rancher/charts/appco-kubernetes-cluster-autoscaler | grep -ie "9.50.1"
        "9.50.1"
```

I'll dump some cycles into this

**Comment by irishgordo** [comment]
This is still broken on v2.13.0-alpha6.

<img width="2297" height="1278" alt="Image" src="https://github.com/user-attachments/assets/a546c78f-4509-45f1-bc91-35994787c3d3" />

**Comment by Vicente-Cheng** [comment]
Hi @irishgordo, 
Did you use this tool before? Which means it might be the regression. 

Could we skip this tool for the remaining testing? Or do we need this for the airgapped testing?
Thanks!

**Comment by irishgordo** [comment]
@Vicente-Cheng - the problem just remains in as a UI issue.

I can try to re-test with the latest RCs.

If it still persists it's not really a major issue as the user can still absolutely import a Harvester cluster into Rancher.
The UI just will say that the UI Extension Repo is missing but it just isn't missing.

**Comment by irishgordo** [comment]
The user will always need for an airgapped env to pull down the latest UI plugin catalog, we additionally for testing, need to grab our https://hub.docker.com/r/rancher/ui-extension-harvester-ui-extension/tags image as well https://docs.harvesterhci.io/v1.7/airgap#harvester-ui-extension-with-rancher-integration

**Comment by irishgordo** [comment]
@Vicente-Cheng @a110605 
This seems to look good on v1.7.0-rc2 w/ Harvester UI Extension v1.7.0-rc2 w/ Rancher v2.13.0-rc3.
I will close this out.

<img width="2556" height="1972" alt="Image" src="https://github.com/user-attachments/assets/7161b330-acf8-4057-bfe5-ddd0d216af09" />
<img width="2546" height="1906" alt="Image" src="https://github.com/user-attachments/assets/3c7b7200-1f60-4080-a2e7-40f0f00dd95f" />
<img width="2079" height="1914" alt="Image" src="https://github.com/user-attachments/assets/86147632-ed27-4026-be3d-b55b8af2a310" />
<img width="2127" height="1892" alt="Image" src="https://github.com/user-attachments/assets/e1d31703-4d88-40c3-96f4-cdbc38621c76" />
<img width="2134" height="1983" alt="Image" src="https://github.com/user-attachments/assets/0db8c9e6-0c22-4a93-8c9a-10d7babb4ebc" />

**Comment by irishgordo** [comment]
@a110605 I'm re-opening this, this is present in the process of:
1. have airgapped rancher v2.12.5
2. have `rancher/ui-extension-harvester-ui-extension:1.7.1-rc3` & `rancher/ui-plugin-catalog:4.4.0` present 
3. install harvester ui extension v1.6.1 w/ harvester v1.6.1
4. upgrade Rancher to v2.13.1
5. upgrade Harvester to v1.7.1-rc3
6. add the ui-extension-harvester-ui-extension:1.7.1-rc3 to the catalogs
7. refresh the repo as that just always fails for some reason, but we mention this in the docs
8. wait for refresh to go through and be happy
9. upgrade harvester-ui-extension to v1.7.1-rc3
10. still it will say, on Virtualization Management page that the `The Harvester UI Extension repository is missing` , but it's installed and it does work

<img width="1959" height="1923" alt="Image" src="https://github.com/user-attachments/assets/80303974-70a8-46e9-8262-e0649f031356" />
<img width="1966" height="1921" alt="Image" src="https://github.com/user-attachments/assets/7668bd49-8e72-4a17-93ce-d43c16582541" />
<img width="2507" height="1922" alt="Image" src="https://github.com/user-attachments/assets/488e1e08-019e-4ace-9c53-11a84339019d" />
<img width="2492" height="1919" alt="Image" src="https://github.com/user-attachments/assets/8864f385-24a1-468b-8023-dfa7a8b80b39" />
<img width="2495" height="1911" alt="Image" src="https://github.com/user-attachments/assets/bf2c7335-ab51-4a2c-aadc-576db09ea1bc" />
<img width="2560" height="1928" alt="Image" src="https://github.com/user-attachments/assets/e11e87a3-de45-4847-a5db-96bd10d5d3be" />
<img width="2492" height="1944" alt="Image" src="https://github.com/user-attachments/assets/1b2bdfcb-e9bb-4bff-a4a5-a2640d501c29" />
<img width="2498" height="1897" alt="Image" src="https://github.com/user-attachments/assets/2790e7e6-6d7f-4844-a7c0-a0c200f19b1b" />

**Comment by Vicente-Cheng** [comment]
Hi @irishgordo, 
If you thought this issue is the regression, could you open a new issue for it?
Reopen this might confuse the user because v1.7.0 was released

**Comment by irishgordo** [comment]
Thanks for that mention @Vicente-Cheng , I'll close this out in favor of [10006](https://github.com/harvester/harvester/issues/10006)
